lib/puppet/pops/issues.rb

```diff
@@ -383,7 +383,7 @@ def self.hard_issue(issue_code, *args, &block)
   # Issues when expressions that are not implemented or activated in the current version are used.
   #
   UNSUPPORTED_EXPRESSION = issue :UNSUPPORTED_EXPRESSION do
-    _("Expressions of type %{expression} are not supported in this version of Puppet") % { expression: label.a_an(semantic) }
+    _("Expressions of type %{expression_type} are not supported in this version of Puppet.") % { expression_type: label(semantic) }
```

Comment:
Not sure that reads well for any `expression_type` (they are defined in `model_label_provider.rb`)

Reply:
The suggestions above: "Unsupported range in %{expression_type} expression. Was given %{count} where a maximum of 1 is accepted." would for example generate
"Unsupported range in 'apply' expression expression...." - this because the "expression type" typically ends in 'expression', 'Expression', or 'Statement'.